src/extensions/LensExtension.js

```diff
@@ -0,0 +1,129 @@
+ * @param {Object} opts
+ * @param {boolean=} opts.isLensOn Whether or not to use the lens.
+ * @param {number=} opts.lensSelection Numeric index of the channel to be focused on by the lens.
+ * @param {number=} opts.lensRadius Pixel radius of the lens (default: 100).
+ * @param {Array.<number>=} opts.lensBorderColor RGB color of the border of the lens (default [255, 255, 255]).
+ * @param {number=} opts.lensBorderRadius Percentage of the radius of the lens for a border (default 0.02).
+ * */
+
+export default class LensExtension extends LayerExtension {
+  getShaders() {
+    return {
+      ...super.getShaders(),
+      modules: [lens]
+    };
+  }
+
+  initializeState(context, extension) {
+    const layer = this.getCurrentLayer();
+    // eslint-disable-next-line no-param-reassign
+    const onMouseMove = () => {
+      const { viewportId } = layer.props;
+      const { lensRadius = 100 } = extension.opts;
+      // If there is no viewportId, don't try to do anything.
+      if (!viewportId) {
+        layer.setState({ unprojectLensBounds: [0, 0, 0, 0] });
+        return;
+      }
+      const { mousePosition } = layer.context;
+      const layerView = layer.context.deck.viewManager.views.filter(
+        view => view.id === viewportId
+      )[0];
+      const viewState = layer.context.deck.viewManager.viewState[viewportId];
+      const viewport = layerView.makeViewport({
+        ...viewState,
+        viewState
+      });
+      // If the mouse is in the viewport and the mousePosition exists, set
+      // the state with the bounding box of the circle that will render as a lens.
+      if (mousePosition && viewport.containsPixel(mousePosition)) {
+        const offsetMousePosition = {
+          x: mousePosition.x - viewport.x,
+          y: mousePosition.y - viewport.y
+        };
+        const mousePositionBounds = [
+          // left
+          [offsetMousePosition.x - lensRadius, offsetMousePosition.y],
+          // bottom
+          [offsetMousePosition.x, offsetMousePosition.y + lensRadius],
+          // right
+          [offsetMousePosition.x + lensRadius, offsetMousePosition.y],
+          // top
+          [offsetMousePosition.x, offsetMousePosition.y - lensRadius]
+        ];
+        // Unproject from screen to world coordinates.
+        const unprojectLensBounds = mousePositionBounds.map(
+          (bounds, i) => viewport.unproject(bounds)[i % 2]
+        );
+        layer.setState({ unprojectLensBounds });
+      } else {
+        layer.setState({ unprojectLensBounds: [0, 0, 0, 0] });
+      }
+    };
+    if (this.context.deck) {
+      this.context.deck.eventManager.on({
+        pointermove: onMouseMove,
+        pointerleave: onMouseMove,
+        wheel: onMouseMove
+      });
+    }
+    this.setState({ onMouseMove, unprojectLensBounds: [0, 0, 0, 0] });
+  }
+
+  draw(params, extension) {
```

Comment:
As above, I would suggest an early exit for the composite layers. Also, I question whether the lens should be drawn for the `ImageLayer` that is the bottom layer for `MultiscaleImageLayer`.

Reply:
`draw` isn't called on those layers because they are `CompositeLayers`.  I'll prevent the lens from being passed down to the background image though.